src/main/java/com/mraof/minestuck/item/weapon/OnHitEffect.java

```diff
@@ -180,6 +180,32 @@ public interface OnHitEffect
 		}
 	};
 	
+	OnHitEffect SPREADING_KNOCKBACK = (stack, target, attacker) -> {
+		//Attacker's right direction and position
+		double pRightSin = Math.sin(attacker.getYRot() * ((float) Math.PI / 180F) - Math.PI / 2.0);
+		double pRightCos = Math.cos(attacker.getYRot() * ((float) Math.PI / 180F) - Math.PI / 2.0);
+		double pX = attacker.getX(), pZ = attacker.getZ();
+		
+		//Other entity position
+		double eX = target.getX(), eZ = target.getZ();
+		//Direction vector player - entity
+		double dirToEntityX = eX - pX, dirToEntityZ = eZ - pZ;
+		double dirMagnitude = Math.sqrt((dirToEntityX * dirToEntityX + dirToEntityZ * dirToEntityZ));
+		dirToEntityX /= dirMagnitude;
+		dirToEntityZ /= dirMagnitude;
+		//Dot product of other entity direction and player's right
+		double dot = (pRightSin * dirToEntityX) + (pRightCos * dirToEntityZ);
+		double expDot = 1.5 * Math.pow(1 - (Math.abs(dot)), 0.3);
+		//Knockback direction vector
+		double knockbackX = -dirToEntityX + (pRightCos * Mth.sign(dot) * expDot);
+		double knockbackZ = -dirToEntityZ + (pRightSin * Mth.sign(dot) * expDot);
+		dirMagnitude = Math.sqrt((knockbackX * knockbackX + knockbackZ * knockbackZ));
+		knockbackX /= dirMagnitude;
+		knockbackZ /= dirMagnitude;
+		
+		target.knockback(0.7f, knockbackX, knockbackZ);
+	};
```

Comment:
Would it make sense to use some vector objects to make the math here more readable?

Reply:
I was avoiding adding an import to the file. Change made as it is visually better, and does not require the redundant manual math. Same to the vector math used in SuspicionEffect.java.